File: urbanairship-message-center/src/test/java/com/urbanairship/TestApplication.java
/* Copyright Airship and Contributors */

package com.urbanairship;

import android.app.Application;
import android.content.pm.ProviderInfo;

import org.robolectric.Robolectric;
import org.robolectric.TestLifecycleApplication;

import java.lang.reflect.Method;

public class TestApplication extends Application implements TestLifecycleApplication {

    private PreferenceDataStore preferenceDataStore;
    private TestAirshipRuntimeConfig testRuntimeConfig;

    @Override
    public void onCreate() {
        super.onCreate();

        this.preferenceDataStore = new PreferenceDataStore(getApplicationContext());
        testRuntimeConfig = TestAirshipRuntimeConfig.newTestConfig();
        AirshipConfigOptions airshipConfigOptions = testRuntimeConfig.getConfigOptions();

        UAirship.application = this;
        UAirship.isFlying = true;
        UAirship.isTakingOff = true;

        UAirship.sharedAirship = new UAirship(airshipConfigOptions);
        UAirship.sharedAirship.preferenceDataStore = preferenceDataStore;
        UAirship.sharedAirship.runtimeConfig = testRuntimeConfig;
        ProviderInfo info = new ProviderInfo();
        info.authority = UrbanAirshipProvider.getAuthorityString(this);
        Robolectric.buildContentProvider(UrbanAirshipProvider.class).create(info);
    }

    @Override
    public void afterTest(Method method) {
    }

    @Override
    public void beforeTest(Method method) {
    }

    @Override
    public void prepareTest(Object test) {

    }
}

File: urbanairship-test/src/main/java/com/urbanairship/TestApplication.kt
/* Copyright Airship and Contributors */
package com.urbanairship

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import androidx.test.core.app.ApplicationProvider
import java.lang.reflect.Method
import org.robolectric.TestLifecycleApplication

@SuppressLint("VisibleForTests")
public class TestApplication public constructor() : Application(), TestLifecycleApplication {


    override fun onCreate() {
        super.onCreate()
    }

    override fun onTerminate() {
        super.onTerminate()
    }

    override fun beforeTest(method: Method?) {
    }

    override fun prepareTest(test: Any?) {
    }

    override fun afterTest(method: Method?) {
    }

    public companion object {
        public fun getApplication(): TestApplication {
            return ApplicationProvider.getApplicationContext<Context>() as TestApplication
        }
    }
}
